### ISSUE TITLE
CausalForestDML(drate=True): ate_stderr_ / att_stderr_ are too small by sqrt(d_y*d_t) with multiple treatments or outcomes

### ISSUE THREAD

**Issue by lucaluo925** [py-why/EconML#1066]
With `discrete_treatment=True, drate=True` and 3+ treatment arms (or more than one outcome), `ate_stderr_`, `att_stderr_`, `ate__inference()` and `att__inference()` report standard errors that are too small by exactly sqrt(d_y · d_t).

In `econml/dml/causal_forest.py`, `_CausalForestFinalWrapper._ate_and_stderr`:

```python
nonnan = np.sum(~np.isnan(drpreds))
stderr = (np.nanstd(drpreds, axis=0) / np.sqrt(nonnan)).reshape(...)
```

`nonnan` counts every element of the (n, d_y, d_t) array instead of the non-NaN rows per (y, t) column, so the standard deviation is divided by sqrt(n · d_y · d_t) instead of sqrt(n). The ATT masks inherit the same issue. The binary-treatment, single-outcome case is correct (d_y · d_t = 1), which is presumably why this hasn't shown up; the existing tests only access these attributes without checking values.

Impact: in a seeded simulation (60 runs, n=1000, 3 arms, true ATEs 1 and 2), nominal 95% intervals built from the reported `ate_stderr_` covered the true ATE only 82% / 77% of the time; with the per-column sqrt(n) standard error the coverage was 95% / 90%.

Suggested fix: `nonnan = np.sum(~np.isnan(drpreds), axis=0)`; the rest of the line then broadcasts element-wise. Happy to open a PR with a test comparing `ate_stderr_` / `att_stderr_(T=t)` to `nanstd/sqrt(count)` of the pseudo-outcomes for a 3-arm treatment with one and two outcomes.

AI disclosure: I found this with the help of an AI assistant (Claude), which also ran the reproduction below and drafted this issue; I reviewed it before posting.

```python
# CausalForestDML(discrete_treatment=True, drate=True): ate_stderr_/att_stderr_ divide by
# sqrt(number of non-NaN ELEMENTS of drpreds (n*d_y*d_t)) instead of sqrt(n per column).
import numpy as np, warnings; warnings.filterwarnings("ignore")
import econml; from econml.dml import CausalForestDML
from sklearn.linear_model import LinearRegression, LogisticRegression
print("econml", econml.__version__)
rng = np.random.default_rng(0); n = 2000
X = rng.normal(size=(n, 2)); W = rng.normal(size=(n, 2))
T = rng.integers(0, 3, size=n)                       # 3 arms -> d_t = 2
Y = 1.0 * (T == 1) + 2.0 * (T == 2) + X[:, 0] + W[:, 0] + rng.normal(size=n)
for Yfit, label in [(Y, "d_y=1"), (np.column_stack([Y, Y + rng.normal(size=n)]), "d_y=2")]:
    est = CausalForestDML(model_y=LinearRegression(), model_t=LogisticRegression(),
                          discrete_treatment=True, drate=True, n_estimators=200, random_state=0)
    est.fit(Yfit, T, X=X, W=W)
    dr = est.rlearner_model_final_._oob_preds          # doubly-robust pseudo-outcomes, shape (n, d_y, d_t)
    k = (~np.isnan(dr)).sum(axis=0)                    # per-(y,t) count of usable rows
    expected = np.nanstd(dr, axis=0) / np.sqrt(k)      # closed-form SE of a sample mean
    print(label, "ate_", np.round(est.ate_, 3))
    print("  reported ate_stderr_:", np.round(est.ate_stderr_, 4))
    print("  expected std/sqrt(n):", np.round(expected.squeeze(), 4))
    print("  ratio expected/reported:", np.round(expected.squeeze() / est.ate_stderr_, 3))
    m = T == 1
    exp_att = np.nanstd(dr[m], axis=0) / np.sqrt((~np.isnan(dr[m])).sum(axis=0))
    print("  att_stderr_(T=1) reported:", np.round(est.att_stderr_(T=1), 4), "expected:", np.round(exp_att.squeeze(), 4))
# Binary single-outcome control case (d_t=1,d_y=1) is correct:
Tb = (T > 0).astype(int)
est = CausalForestDML(model_y=LinearRegression(), model_t=LogisticRegression(),
                      discrete_treatment=True, drate=True, n_estimators=200, random_state=0).fit(Y, Tb, X=X, W=W)
dr = est.rlearner_model_final_._oob_preds
print("binary control: reported", np.round(est.ate_stderr_, 4), "expected", np.round(np.nanstd(dr) / np.sqrt((~np.isnan(dr)).sum()), 4))
```

Output (econml 0.17.0 from PyPI, scikit-learn 1.9.1, numpy 2.4.6, Python 3.11; the function is unchanged on main f0fc2e7):
```
econml 0.17.0
d_y=1 ate_ [0.977 2.04 ]
  reported ate_stderr_: [0.0344 0.0334]
  expected std/sqrt(n): [0.0486 0.0473]
  ratio expected/reported: [1.414 1.414]
  att_stderr_(T=1) reported: [0.0861 0.0432] expected: [0.1217 0.0611]
d_y=2 ate_ [[0.977 2.054]
 [0.95  2.022]]
  reported ate_stderr_: [[0.0243 0.0346]
 [0.0242 0.0339]]
  expected std/sqrt(n): [[0.0486 0.0691]
 [0.0483 0.0679]]
  ratio expected/reported: [[2. 2.]
 [2. 2.]]
  att_stderr_(T=1) reported: [[0.0609 0.0861]
 [0.0315 0.0426]] expected: [[0.1217 0.1723]
 [0.063  0.0852]]
binary control: reported [0.0502] expected 0.0502
```
